### PR TITLE
Feature/docker test

### PR DESCRIPTION
**Pull request by VikrantKS**
# Issue Link

https://github.com/LambdaTest/test-at-scale/issues/23
# Description

This PR includes
- Docker test cases 
- Fix stop and removal of spawned  docker containers if synapse stops gracefully 
Note : We cannot remove spawned docker containers if synapse is killed forcefully 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally
- [x] Tested with unit test

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/runner/docker/docker.go [hunk 1]
```diff
@@ -94,11 +112,24 @@ func (d *docker) Create(ctx context.Context, r *core.RunnerOptions) core.Contain
 }
 
 func (d *docker) Destroy(ctx context.Context, r *core.RunnerOptions) error {
-	if err := d.client.ContainerStop(ctx, r.ContainerID, nil); err != nil {
+	gracefulyContainerStopDuration := time.Second * 10
```
Comment (rishabh-arya95) [inline pkg/runner/docker/docker.go:115]: make it const

**Review by rishabh-arya95 — COMMENTED** [review]
only minor comments, rest lgtm

**Review by saurabh-prakash — APPROVED** [review]

File: cmd/synapse/bin.go [hunk 2]
```diff
@@ -142,3 +145,11 @@ func run(cmd *cobra.Command, args []string) {
 	}
 
 }
+
+func setEnv() {
+	os.Setenv(global.AutoRemoveEnv, strconv.FormatBool(true))
+	os.Setenv(global.LocalEnv, strconv.FormatBool(true))
+	os.Setenv(global.SynapseHostEnv, utils.GetOutboundIP())
+	os.Setenv(global.NetworkEnvName, "test-at-scale")
```
[the remark this answers is not in the document]
Comment (VikrantKS) [reply cmd/synapse/bin.go]: moved to global constant

File: pkg/runner/docker/docker.go (hunk 1 above)
Comment (VikrantKS) [reply pkg/runner/docker/docker.go]: making it package level variable as the ContainerStop() expects pointer

File: cmd/synapse/bin.go [hunk 3]
```diff
@@ -142,3 +145,11 @@ func run(cmd *cobra.Command, args []string) {
...
+
```
[the remark this answers is not in the document]
Comment (VikrantKS) [reply cmd/synapse/bin.go]: Noted and changed

**Review by rishabh-arya95 — APPROVED** [review]

**Review by saurabh-prakash — APPROVED** [review]

**Pull request merged**